oomph/c_output.py

```diff
@@ -446,6 +481,47 @@ def emit_toplevel_statement(self, top_statement: tast.ToplevelStatement) -> str:
                 )
             )
 
+        if isinstance(top_statement, tast.UnionDef):
+            assert top_statement.type.types is not None
+            name = self.get_type_c_name(top_statement.type)
+            return (
+                # Underlying C union
+                f"union union_{name} {{\n"
+                + "".join(
+                    f"\t{self.emit_type(the_type)} item{index};\n"
+                    for index, the_type in enumerate(top_statement.type.types)
+                )
+                + "};\n\n"
+                # Struct to also remember which member is active
+                + f"struct class_{name} {{ union union_{name} val; short membernum; }};\n\n"
```

Comment:
I'm a bit concerned about what `incref()` will do with this. Also, there's no destructor.

Currently non-reference-counted types are not supported in unions at all, so I don't need to worry about them. There's an assert in `types.py`.